expression/builtin_time.go

```diff
@@ -2464,94 +2464,113 @@ type convertTzFunctionClass struct {
 	baseFunctionClass
 }
 
+func (c *convertTzFunctionClass) getDecimal(ctx context.Context, arg Expression) int {
+	decimal := 6
+	if dt, isConstant := arg.(*Constant); isConstant {
+		switch fieldTp2EvalTp(arg.GetType()) {
+		case tpInt:
+			decimal = 0
+		case tpReal, tpDecimal:
+			decimal = arg.GetType().Decimal
+		case tpString:
+			str, isNull, err := dt.EvalString(nil, ctx.GetSessionVars().StmtCtx)
```

Comment:
Is it safe to ignore this error?

Reply:
yes. this happens during type inference, here we only need to calculate the `Decimal` for return type. This parameter will be calculated again during the execution stage, if any error detected, it will be returned.